bolt-modules/boltlib/lib/puppet/functions/upload_file.rb

```diff
@@ -47,18 +47,19 @@
     return_type 'ResultSet'
   end
 
-  def upload_file(scope, source, destination, targets, options = nil)
+  def upload_file(scope, source, destination, targets, options = {})
     upload_file_with_description(scope, source, destination, targets, nil, options)
   end
 
-  def upload_file_with_description(scope, source, destination, targets, description = nil, options = nil)
+  def upload_file_with_description(scope, source, destination, targets, description = nil, options = {})
     unless Puppet[:tasks]
       raise Puppet::ParseErrorWithIssue
         .from_issue_and_stack(Bolt::PAL::Issues::PLAN_OPERATION_NOT_SUPPORTED_WHEN_COMPILING, action: 'upload_file')
     end
 
-    options ||= {}
-    options = options.merge('_description' => description) if description
+    options = options.select { |opt| opt.start_with?('_') }.map { |k, v| [k.gsub(/^_/, '').to_sym, v] }.to_h
```

Comment:
Why do you use `gsub` here? Looks like for `run_plan` and this you use it, but for the others its just `sub`? Are there multiple underscores to replace for only these two?

Reply:
I think I was just inconsistent. I'll standardize them all the `sub`.